src/sdl2/render.rs

```diff
@@ -252,24 +256,25 @@ impl<T> Deref for Canvas<T> {
 /// Its context may be shared with the `TextureCreator`.
 ///
 /// The context will not be dropped until all references of it are out of scope.
-pub struct Canvas<T> {
+pub struct Canvas<T, TC> {
     target: T,
-    context: Rc<RendererContext>,
+    context: Rc<RendererContext<TC>>,
 }
 
 /// Alias for a `Canvas` that was created out of a `Surface`
-pub type SurfaceCanvas<'s> = Canvas<Surface<'s>>;
+pub type SurfaceCanvas<'s> = Canvas<Surface<'s>, SurfaceContext<'s>>;
 
 /// Methods for the `SurfaceCanvas`.
-impl<'s> Canvas<Surface<'s>> {
+impl<'s> Canvas<Surface<'s>, SurfaceContext<'s>> {
```

Comment:
Yeah right, it's more complex than I thought it would be. The aliases will be a great help here to make all of this more readable.